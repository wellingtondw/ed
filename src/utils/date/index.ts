type GetFullYearParams = {
  date: string | Date;
};
export const getFullYear = ({ date }: GetFullYearParams) => {
  return new Date(date).getFullYear();
};
