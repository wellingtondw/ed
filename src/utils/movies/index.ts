import { MovieCardListDTO } from 'components/MovieCardList';

export type MoviesDTO = {
  poster_path: string;
  title: string;
  release_date: string;
  vote_average: number;
  id: number;
};

type FormatMoviesListParams = {
  items: MoviesDTO[];
};

export const formatMoviesList = ({ items }: FormatMoviesListParams) => {
  return items.map((item): MovieCardListDTO => {
    const { poster_path, title, release_date, vote_average, id } = item;

    return {
      posterImage: poster_path,
      title,
      date: release_date,
      rating: vote_average,
      id
    };
  });
};
