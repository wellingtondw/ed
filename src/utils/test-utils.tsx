import { createStore } from 'redux';
import { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { createMemoryHistory } from 'history';

import { ThemeProvider } from 'styled-components';
import theme from 'styles/theme';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import rootReducer from '../store/modules/rootReducer';

const memoryHistory = createMemoryHistory();

const customRender = (
  ui: ReactElement,
  initialState = {},
  store = createStore(rootReducer, initialState),
  history = memoryHistory,
  { ...renderOptions }: RenderOptions = {}
) =>
  render(
    <Provider store={store}>
      <Router navigator={history} location={history.location}>
        <ThemeProvider theme={theme}>{ui}</ThemeProvider>
      </Router>
    </Provider>,
    renderOptions
  );

export * from '@testing-library/react';
export { customRender as render };
