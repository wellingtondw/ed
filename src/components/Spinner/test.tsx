import { render } from 'utils/test-utils';
import { Spinner } from '.';

describe('<Spinner />', () => {
  it('should be able to render correctly', () => {
    const { container } = render(<Spinner />);

    expect(container).toMatchSnapshot();
  });
});
