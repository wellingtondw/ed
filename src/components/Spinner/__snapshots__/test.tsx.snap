// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should be able to render correctly 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  width: 80px;
  height: 13px;
}

.c0 div {
  position: absolute;
  top: 0px;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background: #835AFD;
  -webkit-animation-timing-function: cubic-bezier(0,1,1,0);
  animation-timing-function: cubic-bezier(0,1,1,0);
}

.c0 div:nth-child(1) {
  left: 8px;
  -webkit-animation: dTWfpt 0.6s infinite;
  animation: dTWfpt 0.6s infinite;
}

.c0 div:nth-child(2) {
  left: 8px;
  -webkit-animation: BVSOq 0.6s infinite;
  animation: BVSOq 0.6s infinite;
}

.c0 div:nth-child(3) {
  left: 32px;
  -webkit-animation: BVSOq 0.6s infinite;
  animation: BVSOq 0.6s infinite;
}

.c0 div:nth-child(4) {
  left: 56px;
  -webkit-animation: ceDJoN 0.6s infinite;
  animation: ceDJoN 0.6s infinite;
}

<div>
  <div
    class="c0"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
